### PR TITLE
Implement Send Alert Text tests for WPT wdspec tests

### PR DESCRIPTION
**Pull request by alchiem**
Testing webdriver send alert text spec.
http://www.w3.org/TR/webdriver/#send-alert-text

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/18719)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/278002678)

**Comment by thejohnjansen** [comment]
Can anyone on here (@andreastt, @shs96c) approve this one? It's been a while, and I'd love to see it merged before tpac.

File: webdriver/tests/user_prompts/send_alert_text.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+
+def send_alert_text(session, body=None):
+    return session.transport.send("POST", "session/{session_id}/alert/text"
+                                  .format(session_id=session.session_id), body)
+
+# Used to check if alert text sent was correct
+def accept_alert(session):
+    return session.transport.send("POST", "session/{session_id}/alert/accept"
+                                  .format(session_id=session.session_id))
```
Comment (andreastt) [inline webdriver/tests/user_prompts/send_alert_text.py]: You don’t have to make a raw call to this endpoint. We only do that for the command under test, which is Send Alert Text.

File: webdriver/tests/user_prompts/send_alert_text.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+# 18.4 Send Alert Text
+
+def test_no_browsing_context(session, create_window):
+    # 18.4 step 1
+    session.window_handle = create_window()
+    session.close()
+
+    body = {
+        "text": "Federer"
+        }
```
Comment (andreastt) [inline webdriver/tests/user_prompts/send_alert_text.py]: Can you please pull all the bodies throughout the test onto a single line?

File: webdriver/tests/user_prompts/send_alert_text.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
+from tests.support.asserts import assert_error, assert_success
+
+
+def send_alert_text(session, body=None):
```
Comment (andreastt) [inline webdriver/tests/user_prompts/send_alert_text.py]: Can you please test various forms of invalid input to the Send Alert Text command?

File: webdriver/tests/user_prompts/send_alert_text.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    body = {
+        "text": "Federer"
+        }
+    response = send_alert_text(session, body)
+    assert_error(response, "element not interactable")
+
+
+def test_confirm_element_not_interactable(session):
+    # 18.4 step 3
+    session.execute_script("window.confirm(\"Hello\");")
```
Comment (andreastt) [inline webdriver/tests/user_prompts/send_alert_text.py]: You will probably also want to test `window.print` et al., although if you’re pressed on time I’m happy for that to be addressed in a follow-up later.

File: webdriver/tests/user_prompts/send_alert_text.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    # 18.4 step 6
+    session.execute_script("window.result = window.prompt(\"Enter Your Name: \", \"Name\");")
+    body = {
+        "text": "Federer"
+        }
+    send_response = send_alert_text(session, body)
+    assert_success(send_response)
+    accept_response = accept_alert(session)
+    assert_success(accept_response)
+    assert session.execute_script("return window.result") == "Federer"
```
Comment (andreastt) [inline webdriver/tests/user_prompts/send_alert_text.py]: Can you test a string with whitespace, please?

**Review by andreastt — CHANGES_REQUESTED** [review]
Sorry, I had an unpublished review for this.

File: webdriver/tests/user_prompts/send_alert_text.py (hunk 3 above)
Comment (alchiem) [reply webdriver/tests/user_prompts/send_alert_text.py]: Could you provide me with some examples of invalid input? Thank you

File: webdriver/tests/user_prompts/send_alert_text.py (hunk 1 above)
Comment (alchiem) [reply webdriver/tests/user_prompts/send_alert_text.py]: fixed

File: webdriver/tests/user_prompts/send_alert_text.py (hunk 2 above)
Comment (alchiem) [reply webdriver/tests/user_prompts/send_alert_text.py]: fixed

File: webdriver/tests/user_prompts/send_alert_text.py (hunk 4 above)
Comment (alchiem) [reply webdriver/tests/user_prompts/send_alert_text.py]: In the spec, it says "The Send Alert Text command sets the text field of a window.prompt user prompt to the given value." 

Should window.print be included in this test?

File: webdriver/tests/user_prompts/send_alert_text.py (hunk 5 above)
Comment (alchiem) [reply webdriver/tests/user_prompts/send_alert_text.py]: fixed

File: webdriver/tests/user_prompts/send_alert_text.py (hunk 3 above)
Comment (andreastt) [reply webdriver/tests/user_prompts/send_alert_text.py]: Null, undefined, number, object, array

File: webdriver/tests/user_prompts/send_alert_text.py (hunk 4 above)
Comment (andreastt) [reply webdriver/tests/user_prompts/send_alert_text.py]: This is fine.

**Review by andreastt — APPROVED** [review]

**Comment by alchiem** [comment]
Since you merged it already. Do I just create another PR for the test for different inputs?

File: webdriver/tests/user_prompts/send_alert_text.py (hunk 3 above)
Comment (alchiem) [reply webdriver/tests/user_prompts/send_alert_text.py]: fixed

**Comment by andreastt** [comment]
@alchiem Please feel free to do that.

**Pull request merged**